frontend/src/app/dataset/dataset.component.ts

```diff
@@ -245,6 +277,7 @@ export class DatasetComponent {
             .set(channel, true);
         }
         // Turn on the plot that the user requested.
+        console.log('ds togging', channel + this.currentOptions);
```

Comment:
Is the console logging too much? You have added a bunch of them. Would it make sense to wrap them under a debug flag or variable?